InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs

```diff
@@ -0,0 +1,96 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using InternTrack.Core.Api.Models.Interns;
+using InternTrack.Core.Api.Models.Interns.Exceptions;
+using Moq;
+using Xunit;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRetrieveIfSqlExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid someInternId = Guid.NewGuid();
+            var sqlException = GetSqlException();
+
+            var failedInternStorageException =
+                new FailedInternStorageException(sqlException);
+
+            var expectedInternDependencyException =
+                new InternDependencyException(failedInternStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            ValueTask<Intern> retrieveInternByIdTask =
+                this.internService.RetrieveInternByIdAsync(someInternId);
+
+            // then
+            await Assert.ThrowsAsync<InternDependencyException>(() =>
+                retrieveInternByIdTask.AsTask());
```

Comment:
This is the old way of checking exceptions

Reply:
@ADILATIC here is a [reference.](Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RetrievebyId.cs) You have to look for newer projects within the community. The Standard is ever changing and being improved upon.